projects/igniteui-angular/src/lib/combo/combo.common.ts

```diff
@@ -980,6 +986,18 @@ export abstract class IgxComboBaseDirective extends DisplayDensityBase implement
         });
     }
 
+        /** @hidden @internal */
+        public ngDoCheck() {
+            if (this._init) {
+                return;
+            }
+    
+            let changes = this._iterableDiffer.diff(this.data);
```

Comment:
I don't think we actually need to be invoking the IterableDiffer API as the only thing it's doing right now is it is checking if we have populated data. Which you can easily do by just checking `this.data?.length`, which you had already done in the previous change. We can also check for `this.selection` just to be sure that we actually have some selection before calling `createDisplayText`. In any case all of this is going to prevent `createDisplayText` from being called a few times before the data has loaded, so it's more of an optimization check.